src/routes/download/download.router.ts

```diff
@@ -0,0 +1,56 @@
+import { Request, Response, Router } from 'express';
+import { get } from 'lodash';
+import JSZip from 'jszip';
+import moment from 'moment';
+
+import { DOWNLOAD_SEQUENCES_ENDPOINT } from '../../constants/endpoint';
+import { downloadSequenceFiles } from './download.service';
+import logger from '../../logger';
+
+export const router: Router = Router();
+
+router.post(DOWNLOAD_SEQUENCES_ENDPOINT, async (req: Request, res: Response) => {
+  const ids = get(req.body, 'ids', []);
+  console.time('zip download');
+  try {
+    const assets = await downloadSequenceFiles(ids);
+    const zip = new JSZip();
+    await assets.map(async (asset) => {
+      const sequenceId = (await asset).sequenceId;
+      // writing the files to a separate folder for each sequence id
+      // if there are no files for an individual id, the folder is created but will be empty so the user is aware
+      zip.folder(sequenceId);
+      await Promise.all(
+        asset.files.map((file) => {
+          if (file) {
+            logger.info(`Writing file ${file.name} to zip folder ${sequenceId}`);
+            zip.file(`${sequenceId}/${file.name}`, file.stream);
+          }
+        }),
+      );
+    });
+    const zipName = `sequence_files_${moment().format('YYYY-MM-DDTHH:mm')}.zip`;
+    res.set('Content-Type', 'application/zip');
+    res.attachment(zipName);
+    res.set('Access-Control-Expose-Headers', 'Content-Disposition');
+    zip
+      .generateNodeStream({ type: 'nodebuffer', streamFiles: true })
+      .pipe(res)
+      .on('error', (err) => {
+        logger.info(`Error in zip stream: ${err}`);
+        res.status(500).write(err);
+      })
+      .on('finish', () => {
+        logger.info(`Zip completed, sending response.`);
+        console.timeEnd('zip download');
+        res.status(200).send();
+      });
+  } catch (error) {
+    logger.error(`Error downloading zip file: ${error}`);
```

Comment:
This catch block is around many things that could fail such as the download sequence files, http response interactions, zip writing... 

Maybe we should make the log message more generi, like "Error handling sequence file download"